state-sync/diem-data-client/src/diemnet/tests.rs

```diff
@@ -363,11 +352,74 @@ async fn bad_peer_is_eventually_banned_callback() {
         .advertised_data
         .transactions
         .contains(&CompleteDataRange::new(0, 200).unwrap()));
+}
 
-    // we should still be able to send the good peer a request
-    let response = client
-        .get_transactions_with_proof(100, 50, 100, false)
-        .await
-        .unwrap();
-    assert_eq!(response.payload, TransactionListWithProof::new_empty());
+#[tokio::test]
+async fn bad_peer_is_eventually_added_back() {
+    ::diem_logger::Logger::init_for_testing();
+    let (mut mock_network, mock_time, client, poller) = MockNetwork::new();
+
+    // add a connected peer
+    mock_network.add_connected_peer();
+
+    tokio::spawn(poller.start());
+    tokio::spawn(async move {
+        while let Some((_, _, request, response_sender)) = mock_network.next_request().await {
+            match request {
+                StorageServiceRequest::GetTransactionsWithProof(_) => {
+                    response_sender.send(Ok(StorageServiceResponse::TransactionsWithProof(
+                        TransactionListWithProof::new_empty(),
+                    )))
+                }
+                StorageServiceRequest::GetStorageServerSummary => response_sender.send(Ok(
+                    StorageServiceResponse::StorageServerSummary(mock_storage_summary(200)),
+                )),
+                _ => (),
+            }
+        }
+    });
+
+    // Advance time so the poller sends a data summary request
+    tokio::task::yield_now().await;
+    mock_time.advance_async(DATA_SUMMARY_POLL_INTERVAL).await;
+
+    // Initially this request range is serviceable by this peer
+    let global_summary = client.get_global_data_summary();
+    assert!(global_summary
+        .advertised_data
+        .transactions
+        .contains(&CompleteDataRange::new(0, 200).unwrap()));
+
+    // Keep decreasing this peer's score by considering its responses bad
```

Comment:
nit: I find it clearer when multi-line comments use fullstops. Otherwise, it can be annoying to read. 😄

Reply:
Updated this multi-line comment (and other comments in this file) to use fullstops.